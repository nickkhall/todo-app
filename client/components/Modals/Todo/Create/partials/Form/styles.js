export default {
  formField: {
    width: '100%'
  }
};
