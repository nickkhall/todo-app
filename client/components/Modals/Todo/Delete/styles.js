// MUI Colors
import red from '@material-ui/core/colors/red';

export default {
  deleteTodoModal: {
    display: 'flex',
    flexDirection: 'column'
  },
  todoName: {
    color: red[500],
    fontStyle: 'bold'
  }
};
