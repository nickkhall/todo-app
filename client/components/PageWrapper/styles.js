export default () => ({
  pageWrapper: {
    flex: 1,
    display: 'flex',
    maxWidth: '100%',
    margin: 'auto',
    textAlign: 'center',
    maxHeight: '100%'
  }
});
