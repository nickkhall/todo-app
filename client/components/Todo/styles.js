// Colors
import red from '@material-ui/core/colors/red';
import green from '@material-ui/core/colors/green';
import grey from '@material-ui/core/colors/grey';

export default {
	todo: {
		background: grey[200],
		display: 'flex',
		justifyContent: 'center',
		flexDirection: 'column',
		padding: '20px',
		margin: '10px 0',
		'& > *': { margin: '5px 0' },
		'& span': {
			display: 'inline-block',
			margin: '0 5px'
		}
	},
	todoTitle: {
		textOverflow: 'ellipsis',
		overflow: 'hidden',
		whiteSpace: 'nowrap',
		maxWidth: '400px'
	},
	completed: {
		color: green[500]
	},
	incomplete: {
		color: red[500]}
};
