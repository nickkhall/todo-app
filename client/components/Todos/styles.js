export default {
  todosAside: {
    display: 'flex',
    justifyContent: 'space-between'
  },
  noTodos: {
    padding: '20px 0'
  }
};
