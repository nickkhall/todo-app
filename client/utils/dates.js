export const getPrettyDate = (timestamp) => {
  if (timestamp) {
    let date = new Date(timestamp * 1000).toString();
    date = date.substring(0, date.indexOf('GMT') - 1);

    return date;
  }
  return 'Unknown Date';
};
