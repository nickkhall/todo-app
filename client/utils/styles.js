export const combineStyles = (...styles) => function CombineStyles(theme) {
  const outStyles = styles.map((arg) => {
    // Apply the "theme" object for style functions.
    if (typeof arg === 'function') {
      return arg(theme);
    }
    // Objects need no change.
    return arg;
  });

  return outStyles.reduce((acc, val) => Object.assign(acc, val));
};

export const hexToRGB = (hex, alpha = 1) => {
  let parseString = hex;
  if (hex.startsWith('#')) { parseString = hex.slice(1, 7); }
  if (parseString.length !== 6) { return null; }
  const r = parseInt(parseString.slice(0, 2), 16);
  const g = parseInt(parseString.slice(2, 4), 16);
  const b = parseInt(parseString.slice(4, 6), 16);
  if (Number.isNaN(r) || Number.isNaN(g) || Number.isNaN(b)) { return null; }
  return `rgba(${r}, ${g}, ${b}, ${alpha})`;
};
