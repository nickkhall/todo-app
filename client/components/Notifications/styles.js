// MUI Colors
import amber from '@material-ui/core/colors/amber';
import blue from '@material-ui/core/colors/blue';
import green from '@material-ui/core/colors/green';
import red from '@material-ui/core/colors/red';

// Utils
import { hexToRGB } from 'utils/styles';

// Constants
import { BACKGROUND_OPACITY } from './constants';

export default theme => ({
  wrapper: {
    display: 'flex',
    flexDirection: 'column',
    pointerEvents: 'none', /* Allow click to bleed through to elements below */
    position: 'absolute',
    top: '10px',
    zIndex: 4000,

    /* On larger displays, notifications appear at the bottom left. */
    [theme.breakpoints.up('sm')]: {
      top: 'auto',
      bottom: '10px',
      left: 0,
      minWidth: '400px'
    }
  },
  content: {
    color: '#fff',
    fontSize: '1.1em',
    padding: '20px',
    margin: '10px',
    borderRadius: '3px',
    pointerEvents: 'auto'
  },
  success: {
    backgroundColor: hexToRGB(green[600], BACKGROUND_OPACITY)
  },
  error: {
    backgroundColor: hexToRGB(red[600], BACKGROUND_OPACITY)
  },
  info: {
    backgroundColor: hexToRGB(blue[600], BACKGROUND_OPACITY)
  },
  warning: {
    backgroundColor: hexToRGB(amber[700], BACKGROUND_OPACITY)
  }
});
