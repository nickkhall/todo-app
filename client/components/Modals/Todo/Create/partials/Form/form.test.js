// Utils
import { testCommonComponentAttrs } from 'utils/tests';

// Styles
import styles from './styles';

// Components
import CreateTodoForm from './form';

const mockProps = {
  classes: styles
};

describe('<CreateTodoForm />', () => {
  testCommonComponentAttrs(CreateTodoForm, mockProps);
});
