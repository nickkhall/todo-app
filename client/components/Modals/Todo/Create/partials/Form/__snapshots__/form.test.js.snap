// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateTodoForm /> when rendering should render a component matching the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <TextField
    className={
      Object {
        "width": "100%",
      }
    }
    key="name"
    label="Name"
    name="name"
    onChange={[Function]}
    required={true}
    select={false}
  />
</form>
`;
